src/default-config-backend.cpp

```diff
@@ -9,29 +9,56 @@
 #include <sys/inotify.h>
 #include <unistd.h>
 
-#define INOT_BUF_SIZE (1024 * sizeof(inotify_event))
-static char buf[INOT_BUF_SIZE];
-
+#define INOT_BUF_SIZE (sizeof(inotify_event) + NAME_MAX + 1)
 
 static std::string config_dir, config_file;
 wf::config::config_manager_t *cfg_manager;
 
-static void reload_config(int fd)
+static void readd_watch(int fd)
 {
-    wf::config::load_configuration_options_from_file(*cfg_manager, config_file);
     inotify_add_watch(fd, config_dir.c_str(), IN_CREATE);
     inotify_add_watch(fd, config_file.c_str(), IN_MODIFY);
 }
 
+static void reload_config(int fd)
+{
+    wf::config::load_configuration_options_from_file(*cfg_manager, config_file);
+    readd_watch(fd);
+}
+
 static int handle_config_updated(int fd, uint32_t mask, void *data)
 {
     LOGD("Reloading configuration file");
 
-    /* read, but don't use */
-    read(fd, buf, INOT_BUF_SIZE);
-    reload_config(fd);
+    char buf[INOT_BUF_SIZE]
+    __attribute__((aligned(alignof(inotify_event))));
+
+    bool should_reload = false;
+
+    while (read(fd, buf, INOT_BUF_SIZE) > 0)
```

Comment:
Read is not guaranteed to read INOT_BUF_SIZE bytes, see https://man7.org/linux/man-pages/man2/read.2.html
Thus, we need to handle this situation. Perhaps we need a generic utility function for reading exactly N bytes? https://github.com/WayfireWM/wayfire/blob/ipc/src/ipc/ipc.cpp#L172

Reply:
Do you mean `read` might not read `sizeof(inotify_event)` or `INOT_BUF_SIZE`? `sizeof(inotify_event)` is the size needed, and the rest is just according to the `len` field.

Reply:
I mean that, read might read just a single byte, then get interrupted, right?

Reply:
Right. Would checking the bytes read to be larger than `sizeof(inotify_event)` suffice then? Or would checking `errno` be a better idea?

Reply:
The change was trivial enough that I did it anyway. It now checks for `while (read(...) >= sizeof(inotify_event))`.